sqlglot/tokens.py

```diff
@@ -627,6 +627,8 @@ def tokenize(self, sql):
                 if white_space == TokenType.BREAK:
                     self._col = 1
                     self._line += 1
+            elif self._char == "0" and self._peek == "x":
```

Comment:
i think this needs to move into the scan_number branch to handle the select 0x -> 0 as x case